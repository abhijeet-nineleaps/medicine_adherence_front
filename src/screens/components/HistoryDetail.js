import React, {useEffect, useState} from 'react';
import {View, Text} from 'react-native';
import Timeline from 'react-native-timeline-flatlist';
import Icon from 'react-native-vector-icons/FontAwesome5'

const HistoryDetail = ({data, modalVisibility}) => {
  const [filterData, setfilterData] = useState([]);
  function findTime(ele) {
    let time = parseInt(ele.split(':')[0]),
      period = ele.split(' ')[1],
      mtitle;
    if (time >= 6 && time <= 12 && period === 'AM') {
      mtitle = 'Morning';
    } else if (time >= 12 && time <= 18 && period === 'PM') {
      mtitle = 'Afternoon';
    } else if (time >= 18 && time <= 24 && period === 'PM') {
      mtitle = 'Night';
    }

    return mtitle;
  }
  useEffect(() => {
    const sArray = [];
    let nottaken = data.notTaken.split(','),
      taken = data.taken.split(',');
    nottaken[0] !== '' &&
      nottaken.map(ele => {
        let mtitle = findTime(ele);
        let nottakenObj = {
          time: ele,
          title: <Text style={{color: 'black'}}>{mtitle}</Text>,
          description: <Text style={{color: 'red'}}>No Taken</Text>,
        };
        sArray.push(nottakenObj);
      });

    taken[0] !== '' &&
      taken.map(ele => {
        let mtitle = findTime(ele);

        let nottakenObj = {
          time: ele,
          title: <Text style={{color: 'black'}}>{mtitle}</Text>,
          description: <Text style={{color: 'green'}}>Taken</Text>,
        };
        sArray.push(nottakenObj);
      });

    setfilterData(sArray);
  }, []);

  return (
    <View
      style={{
        height: '85%',
        width: '90%',
        backgroundColor: 'white',
        marginTop: 80,
        marginBottom: 80,
        flex: 1,
        borderRadius: 20,
      }}>
      <View style={{alignItems: 'flex-end', paddingRight: 10, paddingTop: 10}}>
        <Icon
          size={24}
          name='trash'
          color="#3743ab"
          id='delete'
          onPress={modalVisibility}
        />
      </View>
      <View style={{padding: 12, alignItems: 'center'}}>
        <Text style={{color: 'grey', fontSize: 20, fontWeight: '700'}} testID='dateText' >
          Date - {data}  
        </Text>
      </View>
      <Timeline
        style={{flex: 1, color: 'white', marginLeft: 10}}
        data={filterData}
        circleSize={20}
        circleColor="#4dd0e1"
        lineColor="#4dd0e1"
        listViewStyle={{padding: 10}}
        timeContainerStyle={{minWidth: 52}}
        timeStyle={{
          textAlign: 'center',
          color: 'white',
          backgroundColor: '#4dd0e1',
          padding: 5,
          marginBottom: 80,
          borderRadius: 20,
        }}
        titleStyle={{
          padding: 0,
        }}
        descriptionStyle={{marginTop: 8}}
        isUsingFlatlist={true}
        innerCircle={'dot'}
      />
    </View>
  );
};

export default HistoryDetail;
