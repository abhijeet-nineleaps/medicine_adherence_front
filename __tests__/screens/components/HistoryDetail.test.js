import React from 'react';
import renderer from 'react-test-renderer';
import HistoryDetail from '../../../src/screens/components/HistoryDetail';
import Enzyme from 'enzyme';
import {shallow} from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
Enzyme.configure({adapter: new Adapter()});
describe('Click send image', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<HistoryDetail data={undefined} modalVisibility={undefined} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('test open save button', () => {
    const mockFn = jest.fn();
    const wrapper = shallow(<HistoryDetail modalVisibility={mockFn} />);
    wrapper.find('#delete').simulate('press');
  });
});
