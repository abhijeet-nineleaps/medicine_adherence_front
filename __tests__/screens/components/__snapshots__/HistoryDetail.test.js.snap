// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Click send image renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "borderRadius": 20,
      "flex": 1,
      "height": "85%",
      "marginBottom": 80,
      "marginTop": 80,
      "width": "90%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "paddingRight": 10,
        "paddingTop": 10,
      }
    }
  >
    <Text
      allowFontScaling={false}
      id="delete"
      selectable={false}
      style={
        Array [
          Object {
            "color": "#3743ab",
            "fontSize": 24,
          },
          undefined,
          Object {
            "fontFamily": "FontAwesome5Free-Solid",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {
            "fontWeight": "900",
          },
        ]
      }
    >
      ï‡¸
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "padding": 12,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "grey",
          "fontSize": 20,
          "fontWeight": "700",
        }
      }
      testID="dateText"
    >
      Date - 
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "color": "white",
          "flex": 1,
          "marginLeft": 10,
        },
      ]
    }
  >
    <RCTScrollView
      data={Array []}
      extraData={
        Object {
          "data": Array [],
          "width": 0,
          "x": 0,
        }
      }
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      style={
        Array [
          Object {
            "flex": 1,
          },
          Object {
            "padding": 10,
          },
        ]
      }
      viewabilityConfigCallbackPairs={Array []}
    >
      <View />
    </RCTScrollView>
  </View>
</View>
`;
